### DISCUSSION TITLE
Offchain Resolution

### DISCUSSION THREAD

**Discussion by Anonermous** [wevm/viem#933] [Idea / Feature Request] [answered]
I'm wondering if Viem will ever support offchain name resolution? i.e. I want the address of an ENS name like `coinbase.cb.id`. Ethers makes this really simple. I would love to use the same functionality but with Viem!

**Comment by jxom** [comment, marked as answer]
Already supported 😊

**Reply by Anonermous** [reply]
It seems to work when I use the default `http` transport, but if I add a custom transport a large error is thrown 🤔 

```
error: {
       "docsPath": "/docs/contract/readContract",
       "metaMessages": [
         "This could be due to any of the following:",
         "  - The contract does not have the function \"resolve\",",
         "  - The parameters passed to the contract function may be invalid, or",
         "  - The address is not a contract.",
         ...
```

**Reply by jxom** [reply]
Interesting. Mind providing us with a minimal reproducible example via a separate repo or https://new.viem.sh

**Reply by Anonermous** [reply]
Ok I have an example that appears to be giving me the same error `execution reverted`

https://stackblitz.com/edit/viem-getting-started-6ec5on?file=index.ts

You can see that `vitalik.eth` works as expected, but `coinbase.cb.id` does not.

**Reply by Anonermous** [reply]
Hey @jxom just wondering if you were able to look into this yet?
